src/platform/datapath_raw_xdp_win.c

```diff
@@ -1882,3 +1882,124 @@ CxPlatIoXdpShutdownEventComplete(
         Partition);
     CxPlatDpRawRelease((XDP_DATAPATH*)Partition->Xdp);
 }
+
+_IRQL_requires_max_(PASSIVE_LEVEL)
+QUIC_STATUS
+CxPlatDataPathRssConfigGet(
+    _In_ uint32_t InterfaceIndex,
+    _Outptr_ _At_(*RssConfig, __drv_allocatesMem(Mem))
+        CXPLAT_RSS_CONFIG** RssConfig
+    )
+{
+    QUIC_STATUS Status = QUIC_STATUS_SUCCESS;
+    uint32_t RssConfigSize = 0;
+    XDP_RSS_CONFIGURATION *RawRssConfig = NULL;
+
+    if (RssConfig == NULL) {
+        Status = QUIC_STATUS_INVALID_PARAMETER;
+        goto Error;
+    }
+
+    *RssConfig = NULL;
+
+    HANDLE InterfaceHandle = NULL;
+    HRESULT Result = XdpInterfaceOpen(InterfaceIndex, &InterfaceHandle);
+    if (FAILED(Result)) {
+        QuicTraceLogError(
+            XdpGetRssConfigInterfaceOpenFailed,
+            "[ xdp] Failed to open Xdp interface for index %u. Error %d",
+            InterfaceIndex,
+            Result);
+        Status = Result;
+        goto Error;
+    }
+
+    Result = XdpRssGet(InterfaceHandle, RawRssConfig, &RssConfigSize);
+    if (SUCCEEDED(Result)) {
+        QuicTraceLogError(
+            XdpGetRssConfigSizeFailed,
+            "[ xdp][%p] Failed to get RSS configuration size on IfIndex %u, RssConfigSize %u, Result %d",
+            InterfaceHandle,
+            InterfaceIndex,
+            RssConfigSize,
+            Result);
+        Status = QUIC_STATUS_INTERNAL_ERROR;
+        goto Error;
+    }
+
+    RawRssConfig = CXPLAT_ALLOC_PAGED(RssConfigSize, QUIC_POOL_TMP_ALLOC);
+    if (RawRssConfig == NULL) {
+        QuicTraceEvent(
+            AllocFailure,
+            "Allocation of '%s' failed. (%llu bytes)",
+            "XDP RSS Config",
+            RssConfigSize);
+        Status = QUIC_STATUS_OUT_OF_MEMORY;
+        goto Error;
+    }
+
+    Result = XdpRssGet(InterfaceHandle, RawRssConfig, &RssConfigSize);
+    if (FAILED(Result)) {
+        QuicTraceLogError(
+            XdpGetRssConfigFailed,
+            "[ xdp][%p] Failed to get RSS configuration on IfIndex %u, RssConfigSize %u, Result %d",
+            InterfaceHandle,
+            InterfaceIndex,
+            RssConfigSize,
+            Result);
+        Status = Result;
+        goto Error;
+    }
+
+    RssConfigSize = sizeof(CXPLAT_RSS_CONFIG) +
+        RawRssConfig->HashSecretKeySize +
+        RawRssConfig->IndirectionTableSize;
+    CXPLAT_RSS_CONFIG* NewRssConfig =
+        (CXPLAT_RSS_CONFIG*)CXPLAT_ALLOC_NONPAGED(
```

Comment:
You used PAGED above and NONPAGED here. I know it doesn't _really_ matter in user mode, but we might adapt this for kernel one day, so we should get it right. I'd say PAGED for both, as this likely wouldn't need to be called at DISPATCH.

Reply:
I was using NONPAGED for the out parameter because the platform shouldn't assume the caller won't use this information at DISPATCH. The internal, temporary, allocations can be PAGED because they'll be freed before returning and this call is only supported at PASSIVE (because of all the allocations).